### PR TITLE
Switch tooling to `uv`, `ruff`, and `just`

### PR DESCRIPTION
**Pull request by bbrown1867**
_Please explain the changes you made here._

This PR updates the repo to use `uv` for Python tooling, instead of `setuptools`. It also uses `just` to automate some commands. Benefits:

- The complexity of managing the `venv` in the `Makefile` goes away, since `uv` handles all of this for us.
- `setup.py` can be consolidated with `pyproject.toml`.
- CI pipeline is simpler with `astral-sh/setup-uv@v3` and `uv run -p` for multi-Python version testing.
- Lock file ensures the same packages are always used, so dev dependencies can be loosened in `pyproject.toml`.

_Does this close any currently open issues?_

No.

_Checklist:_

- [x] [Individual Contributor License Agreement (CLA)](https://spreadsheets.google.com/spreadsheet/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1) signed

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,19 +15,10 @@ jobs:
...
-        run: make wheel
-      - name: Test
-        run: make test
+      - uses: astral-sh/setup-uv@v3
+      - run: uv run -p ${{ matrix.python-version}} black pyfu_usb/ tests/
+      - run: uv run -p ${{ matrix.python-version}} mypy pyfu_usb/ tests/
+      - run: uv run -p ${{ matrix.python-version}} pylint pyfu_usb/ tests/
+      - run: uv run -p ${{ matrix.python-version}} isort pyfu_usb/ tests/
+      - run: uv run -p ${{ matrix.python-version}} pytest tests/
+      - run: uv build -p ${{ matrix.python-version }}
```
Comment (notenti) [inline .github/workflows/test.yml:24]: nit/suggestion: probably can use `ruff` instead of `isort` and `black`, too

**Review by notenti — APPROVED** [review]
🚀

**Review by mdmedley — APPROVED** [review]
LGTM

File: .github/workflows/test.yml (hunk 1 above)
Comment (bbrown1867) [reply .github/workflows/test.yml:24]: Good call, probably can replace `pylint` as well. Will make that change.

**Pull request merged**